pkg/dbtoken/refresher.go

```diff
@@ -35,6 +36,7 @@ type refresher struct {
 
 	m            metrics.Emitter
 	metricPrefix string
+	refreshCount int
```

Comment:
should this be managed atomically?

Reply:
We are not interested in the exact value so it might as well be a bool. Also any race conditions between reading the value and setting to 0 would actually make the heartbeat shut off one cycle sooner in case of an issue that starts right when heartbeat triggers, so I think it's fine like this.

Reply:
I like the refactor to a bool, thank you